src/librustc_mir/transform/check_consts/ops.rs

```diff
@@ -15,9 +17,9 @@ pub trait NonConstOp: std::fmt::Debug {
     /// Whether this operation can be evaluated by miri.
     const IS_SUPPORTED_IN_MIRI: bool = true;
 
-    /// Returns a boolean indicating whether the feature gate that would allow this operation is
-    /// enabled, or `None` if such a feature gate does not exist.
-    fn feature_gate(_tcx: TyCtxt<'tcx>) -> Option<bool> {
+    /// Returns the `Symbol` for the feature gate that would allow this operation, or `None` if
+    /// such a feature gate does not exist.
+    fn feature_gate() -> Option<Symbol> {
```

Comment:
This change would be better as its own independent PR, as then this PR really only does what it advertises in the title.

Reply:
See #70013. This PR is quite old. The main blocker is blessing a bunch of `//~ ERROR` annotations.